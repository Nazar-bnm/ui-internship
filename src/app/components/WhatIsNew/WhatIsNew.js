import React, { Component } from 'react';
import HttpService from '../../service/httpService';


class WhatIsNew extends Component {
  state = {
    items: [],
  }

  test = async () => {
    const userAPI = new HttpService();
    const response = await userAPI.get('http://localhost:4000/what-is-new');
    this.setState({ items: response.data });
  }

  componentDidMount() {
    this.test();
  }

  render() {
    const { items } = this.state;
    const block = items.map((el, index) => (
      <div key={index}>
        <img style={{ display: 'block' }} src={el.photo}/>
        <span>{el.title}</span>
        <span>{el.collection}</span>
      </div>
    ));
    return (
      <div>
        {block}
      </div>
    );
  }
}

export default WhatIsNew;
