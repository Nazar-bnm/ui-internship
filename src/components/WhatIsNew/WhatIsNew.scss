@import '../../scss/global_styles';

.what-is-new {
  .heading__wrapper {
    margin: 20px 0 15px;
    font-size: 1.1rem;
  }

  &__title {
    font-size: 2rem;
  }

  &-wrapper {
    @include flex($justify: center);

    &__item {
      position: relative;
      overflow: hidden;
      transition: transform 300ms linear;

      &:before {
        position: absolute;
        top: 50%;
        left: 50%;
        z-index: 2;
        display: block;
        content: '';
        width: 0;
        height: 0;
        background: rgba(255, 255, 255, 0.2);
        border-radius: 100%;
        transform: translate(-50%, -50%);
        opacity: 0;
      }

      &:hover {
        transform: scale(1.025);

        &::before {
          animation: circle 1s;
        }
      }
    }
  }

  @keyframes circle {
    0% {
      opacity: 1;
    }
    40% {
      opacity: 1;
    }
    100% {
      width: 200%;
      height: 200%;
      opacity: 0;
    }
  }
}

@media screen and (max-width: 480px) {
  .container {
    flex-direction: column;
    align-content: center;
  }
}
