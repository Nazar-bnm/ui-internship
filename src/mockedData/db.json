{
  "employees": [
    {
      "id": 1,
      "first_name": "Sebastian",
      "last_name": "Eschweiler",
      "email": "sebastian@codingthesmartway.com"
    },
    {
      "id": 2,
      "first_name": "Steve",
      "last_name": "Palmer",
      "email": "steve@codingthesmartway.com"
    },
    {
      "id": 3,
      "first_name": "Ann",
      "last_name": "Smith",
      "email": "ann@codingthesmartway.com"
    }
  ],
  "what-is-new" : [
    {
      "id": 1,
      "photo": "https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg",
      "title": "Black and White",
      "collection": "spring collection 2016",
      "category": "women"
    },
    {
      "id": 2,
      "photo": "https://www.next.ua/nxtcms/resource/image/1233600/portrait_ratio1x1/525/525/69ca86774e073288bc792479650a14cc/Jn/all-baby-girl.jpg",
      "title": "Color Summer",
      "collection": "spring collection 2016",
      "category": "kids"
    },
    {
      "id": 3,
      "photo": "https://www.next.ua/nxtcms/resource/image/1915752/portrait_ratio1x1/525/525/1cf45576cf1de5e2bab187220d4ebeb0/ds/g83-mb-sh819338-lm-357.jpg",
      "title": "Vintage For Him",
      "collection": "spring collection 2016",
      "category": "men"
    }
  ],
  "most-popular": [
    {
      "id": 1,
      "image": "https://i.dailymail.co.uk/1s/2018/12/15/12/7458890-6498875-image-a-2_1544876240893.jpg",
      "label": "new",
      "title": "Black dress",
      "price": "99,50$",
      "sizes": "S - M - L"
    },
    {
      "id": 2,
      "image": "https://i.pinimg.com/564x/99/5d/f3/995df3032c828381332af1dea4e5c590.jpg",
      "label": "sale",
      "title": "Red evening dress",
      "price": "179.00$",
      "sizes": "S - M - L"
    },
    {
      "id": 3,
      "image": "https://i.redd.it/grla2qtpa5721.jpg",
      "label": "",
      "title": "Fit & flare dress",
      "price": "99.00$",
      "sizes": "S - M - L"
    }
  ]
}
