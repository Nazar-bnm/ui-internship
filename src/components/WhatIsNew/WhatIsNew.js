import React, { Component } from 'react';
import HttpService from '../../service/HttpService/httpService';
import ItemInfo from './ItemInfo';
import './WhatIsNew.scss';

class WhatIsNew extends Component {
  constructor(props) {
    super(props);
    this.state = {
      items: []
    };
  }

  componentDidMount() {
    this.downloadProducts();
  }

  async downloadProducts() {
    const userAPI = new HttpService();
    const response = await userAPI.get('http://localhost:4000/what-is-new');
    this.setState({ items: response.data });
  }

  render() {
    const { items } = this.state;
    const block = items.map((item) => (
      <ItemInfo key={item.id} item={item} />
    ));
    return (
      <div className="content">
        <h3 className="title">whats new</h3>
        <div className="container">
          {block}
        </div>
      </div>
    );
  }
}

export default WhatIsNew;
