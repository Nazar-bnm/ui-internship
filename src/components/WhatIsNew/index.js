import WhatIsNew from './WhatIsNew';
export default WhatIsNew;
