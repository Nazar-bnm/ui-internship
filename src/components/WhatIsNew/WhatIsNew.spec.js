// import React from 'react';
// import { shallow } from 'enzyme';
// import WhatIsNew from './WhatIsNew';
//
// describe('<WhatIsNew />', () => {
//   let wrapper;
//
//   beforeEach(() => {
//     wrapper = shallow(
//       <WhatIsNew />
//     );
//   });
//
//   test('renders', () => {
//     wrapper = shallow(<WhatIsNew />);
//     expect(wrapper.exists()).toBe(true);
//   });
//
//   test('should match snapshot', () => {
//     expect(wrapper).toMatchSnapshot();
//   });
// });

import React from 'react';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';

import { mount } from 'enzyme';
import WhatIsNew from './WhatIsNew';
import mockedData from '../../mockedDataForTests';

describe('<WhatIsNew />', () => {
  let wrapper;
  const { mockedWhatIsNewSection } = mockedData;

  beforeEach(() => {
    wrapper = mount(<WhatIsNew />);
    const mock = new MockAdapter(axios);

    mock.onGet(`${process.env.BASE_URL}/what-is-new`).reply(200, mockedWhatIsNewSection);
  });

  test('should check `componentDidMount()`', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'getImages');
    instance.componentDidMount();
    expect(instance.getImages).toHaveBeenCalledTimes(1);
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

