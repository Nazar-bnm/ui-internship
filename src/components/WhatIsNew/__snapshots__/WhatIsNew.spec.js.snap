// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WhatIsNew /> should match snapshot 1`] = `
<div
  className="content what-is-new"
>
  <Heading
    className="what-is-new__title}"
    position="center"
    title="What is new?"
  />
  <div
    className="container what-is-new-wrapper"
  />
</div>
`;
